docs/src/concepts/fields.md

```diff
@@ -0,0 +1,141 @@
+# Fields
+
+With a given grid that allows us to define each point uniquely in a high-dimensional space, we abstract the data values to be defined on the grid under the concept `AbstractField`. Following is the type tree of the abstract field and its derived data types.
+
+```@raw html
+<img src="../assets/field_type_tree.svg" width="70%"/>
+```
+
+## Defining a multi-dimensional `Field`
+
+Consider the following example, where we predefined a variable `grid` of type `Chmy.UniformGrid`, similar as in the previous section [Grids](./grids.md). We can now define physical properties on the grid.
+
+When defining a scalar field `Field` on the grid, we need to specify the arrangement of the field values. These values can either be stored at the cell centers of each control volume `Center()` or on the cell vertices/faces `Vertex()`.
+
+```julia
+# Define geometry, architecture..., a 2D grid
+grid = UniformGrid(arch; origin=(-lx/2, -ly/2), extent=(lx, ly), dims=(nx, ny))
+
+# Define pressure as a scalar field
+Pr   = Field(backend, grid, Center())
+```
+
+With the methods `VectorField` and `TensorField`, we can conveniently construct 2-dimensional and 3-dimensional fields, with predefined locations for each field dimension on a staggered grid.
+
+```julia
+# Define velocity as a vector field on the 2D grid
+V = VectorField(backend, grid)
+
+# Define stress as a tensor field on the 2D grid
+τ = TensorField(backend, grid)
+```
+
+!!! tip "Acquiring Locations on the Grid Cell"
+    One could use a convenient getter for obtaining locations of variable on the staggered-grid. Such as `Chmy.location(Pr)` for scalar-valued pressure field and `Chmy.location(τ.xx)` for a tensor field.
+
+### Setup Initial Conditions on `Field`
+
+We could also modify field values in order to set up initial conditions on the values. We first have an empty field variable `C` defined, by using `interior(C)` we can inspect its values being empty equal to zero.
+
+```julia
+C = Field(backend, grid, Center())
+```
+
+```julia
+# Set initial values of the field randomly
+set!(C, grid, (_, _) -> rand())
+
+# Set initial values to 2D Gaussian
+set!(C, grid, (x, y) -> exp(-x^2 - y^2))
+```
+
+```@raw html
+<img src="../assets/field_set_ic_random.png" width="50%"/>
+```
+
+```@raw html
+<img src="../assets/field_set_ic_gaussian.png" width="50%"/>
+```
+
+
+
+A slightly more complex usage involves passing extra parameters to be used for initial conditions setup.
+
+```julia
+# Define a temperature field with values on cell centers
+T = Field(backend, grid, Center())
+
+# Function for setting up the initial conditions on T
+init_incl(x, y, x0, y0, r, in, out) = ifelse((x - x0)^2 + (y - y0)^2 < r^2, in, out)
+
+# Set up the initial conditions with parameters specified
+set!(T, grid, init_incl; parameters=(x0=0.0, y0=0.0, r=0.1lx, in=T0, out=Ta))
```

Comment:
As a curiosity, would this syntax be valid / implemented?

```julia
set!(T, grid, x->init_incl(x...), (0.0, 0.0, 0.1lx, T0, Ta))
```

Reply:
@albert-de-montserrat @luraess I tried it but it doesn't seem to quite work out yet. Got the following error:
![image](https://github.com/PTsolvers/Chmy.jl/assets/82619264/ae6dfcb0-be2e-4044-a63c-0fc3e13d6e88)

Reply:
I see, `set!` may need some changes. I haven't tried to run it, but something on this direction may do the trick (if you want to implement this syntax obviously)

```julia
function set!(f::Field{T,N}, grid::StructuredGrid{N}, fun::F, parameters::NTuple{NP}; discrete=false, async=false) where {T,F,N, NP}
    loc = location(f)
    dst = interior(f)
    backend = KernelAbstractions.get_backend(dst)
    if discrete
        _set_discrete!(backend, 256, size(dst))(dst, grid, loc, fun, parameters...)
    else
        _set_continuous!(backend, 256, size(dst))(dst, grid, loc, fun, parameters...)
    end
    async || KernelAbstractions.synchronize(backend)
    return
end
```

Reply:
The problem here is that the function `init_incl` takes several arguments, and not one tuple. This works already:

```
set!(T, grid, (x...)->init_incl(x...); parameters=(0.0, 0.0, 0.1lx, T0, Ta))
```

If your comment is rather about making parameters part of the signature rather than a keyword, I guess we can make a decision and change the API if it will be useful to you (to do some splatting tricks, I suppose). Then we also need to provide the default empty `Tuple`:

```julia
function set!(f::Field{T,N}, grid::StructuredGrid{N}, fun::F, parameters::NTuple{NP}=(); discrete=false, async=false) where {T,F,N, NP}
    ...
end
```

Reply:
Thank for the info! 

```julia
set!(T, grid, (x...)->init_incl(x...); parameters=(0.0, 0.0, 0.1lx, T0, Ta))```

is already almost like the syntax I suggested :) 

PS: sorry, didn't want to derail @youwuyou review!